scripts/quests/jeuno/In_the_Mood_for_Love.lua

```diff
@@ -0,0 +1,104 @@
+-----------------------------------
+-- In the Mood for Love
+-----------------------------------
+-- Log ID: 3, Quest ID: 69
+-- Odasel  !pos -58 -6 -111
+-- Matoaka !pos -37 -6 -122
+-----------------------------------
+require('scripts/globals/items')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require("scripts/globals/missions")
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local quest = Quest:new(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.IN_THE_MOOD_FOR_LOVE)
+
+quest.reward =
+{
+    gil = 4800,
+    fame = 30,
+    fameArea = xi.quest.fame_area.JEUNO,
+    title = xi.title.PICK_UP_ARTIST,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getCurrentMission(xi.mission.log_id.COP) > xi.mission.id.cop.A_VESSEL_WITHOUT_A_CAPTAIN
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Odasel'] = quest:progressEvent(10035, 0, xi.items.CHAMELEON_DIAMOND),
+
+            onEventFinish =
+            {
+                [10035] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Odasel'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(10037, 0, xi.items.CHAMELEON_DIAMOND)
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.items.CHAMELEON_DIAMOND) then
+                        return quest:progressEvent(10036, 0, xi.items.CHAMELEON_DIAMOND)
+                    end
+                end,
+            },
+            onEventFinish =
+            {
+                [0] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_COMPLETED
+        end,
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Odasel'] = quest:event(10038):replaceDefault(),
+
+            ['Matoaka'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
```

Comment:
This is introducing a forever var that will never be removed
Better idea is set the var after completing the quest
If complete then
    SetVar
End

Then here we check for if complete and 1 on finish set the var to 0. It maybe name it Option. It is optional lol

Reply:
Awesome! I think I got what you're saying. Commit was pushed with update.